### ISSUE TITLE
F.cosine_similarity returns negative values

### ISSUE THREAD

**Issue by 13952522076** [pytorch/pytorch#45309]
## 🐛 Bug

While the defination of cosine similarity indicates the output should be in the range of [0, 1], the F.cosine_similarity may return negative values.

## To Reproduce

import torch
import torch.nn.functional as F
centroids = torch.tensor([[1.0761,  0.5506,  0.1808, -0.1537,  0.6544, -0.1115,  0.3097, -1.0536,
         -0.9395, -0.5482]])
features = torch.tensor([[0.1226, 0.1939, 0.4267, 0.4908, 0.3608, 0.0829, 0.8254, 0.7406, 0.7263,
         0.2091]])
distance = F.cosine_similarity(features,centroids, dim=-1)
print(distance)

## Expected behavior

F.cosine_similarity should return a positive value. The code just returns tensor([-0.2638]).

## Environment

Collecting environment information...
PyTorch version: 1.6.0
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.5 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: Could not collect
CMake version: Could not collect

Python version: 3.6 (64-bit runtime)
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: Quadro RTX 6000
Nvidia driver version: 440.100
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.19.2
[pip3] torch==1.6.0
[pip3] torchvision==0.7.0
[conda] Could not collect
## Additional context

No

**Comment by 13952522076** [comment]
range[-1,1]
